docs/DeveloperGuide.md

```diff
@@ -1,11 +1,461 @@
+If the command is invalid, the `TaskParser` returns an appropriate warning message to the user.  
+
+The `SprintParser` validates the parameters of the command. If the command is valid, it returns the respective `SprintCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `SprintParser` returns an appropriate warning message to the user.  
+
+The `HelpParser` validates the parameters of the command. If the command is valid, it returns the respective `HelpParserCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `HelpParser` returns an appropriate warning message to the user.
+
+
+The subcommand classes `HelpParserCommandXYZ`, `ProjectCommandXYZ`, `MemberCommandXYZ`, `TaskCommandXYZ`, `SprintCommandXYZ` all inherit from an abstract `Command` class, 
+which has an execute function.
+
+The `ParserManager` then returns the command back to `SCRUMptious`, which then executes the command.
+
+A detailed list of the subcommand classes is as follows:
+* Project Commands:  
+    * `CreateProjectCommand`
+    * `ListProjectCommand`
+    * `SelectProjectCommand`
+    * `ViewProjectCommand`
+* Member Commands:
+    * `AddMemberCommand`
+    * `DeleteMemberCommand`
+* Task Commands:
+    * `AddTaskCommand`
+    * `ChangeTaskPriorityCommand`
+    * `DeleteTaskCommand`
+    * `DoneTaskCommand`
+    * `ViewTaskCommand`
+* Sprint Commands:
+    * `AddSprintTaskCommand`
+    * `AllocateSprintTaskCommand`
+    * `CreateSprintCommand`
+    * `DeallocateSprintTaskCommand`
+    * `EditSprintCommand`
+    * `RemoveSprintTaskCommand`
+    * `ViewSprintCommand`
+* Help Commands:
+    * `HelpCommand`
+    * `MemberHelpCommand`
+    * `ProjectHelpCommand`
+    * `SprintHelpCommand`
+    * `StorageHelpCommand`
+    * `TaskHelpCommand`
+### Model Component
+![Figure X: Simplified class diagram for Model Component](./image/developerguide/modelcomponent.png "Storage Component UML")  
+Link: [Model Package](https://github.com/AY2021S1-CS2113T-F11-4/tp/tree/master/src/main/java/seedu/duke/model)  
+The Model
+* Includes four packages namely, Project, Task, Member and Sprint.
+
+Project package
+* Includes a ProjectList to manage the multiple instances of Project created by the user
+* Each instance of Project stores one instance of ProjectBacklog, ProjectMembers and SprintList.
+
+Task  package
+* Includes a ProjectBacklog to manage every Tasks created by the user
+* Task can be allocated to Sprints and can be assigned to Members, 
+
+Member package
+* Includes a ProjectMembers to manage every Members created by the user
+* Member can be assigned with Tasks and can be allocated to Sprints holding those Tasks
+ 
+Sprint package
+* Includes a SprintList to manage every Sprints created by the user
+* Sprint can contain Tasks and Members allocated to those Tasks
+### Storage Component
+![Figure X: Simplified class diagram for Storage Component, Model and json.simple](./image/developerguide/storagecomponent.png "Storage Component UML")  
```

Comment:
Not a big issue, but perhaps `0..*` could be replaced with `*`?
![image](https://user-images.githubusercontent.com/53231968/97395576-355f0680-1920-11eb-9035-47a9377a7d05.png)